syntax/tags.go

```diff
@@ -27,6 +28,9 @@ func parseTag(tag string) tagOptions {
 		if token == varintOption {
 			opts[varintOption] = 1
 			continue
+		} else if token == optionalOption {
+			opts[optionalOption] = 1
+			continue
```

Comment:
Should you forbid optional<T> on non-pointers?

Reply:
This follows the same pattern as the other tags, and only interprets the tag when it's meaningful.  I'll do a separate "strictness" PR that checks alignment between tags and types.